### ISSUE TITLE
search: cursor close

### ISSUE THREAD

**Issue by pmwkaa** [tarantool/tarantool#1104]
Too many cursors waiting for being closed after use.

stats from box.sophia()
```
performance.tx_active_ro = 431 # number of unclosed cursors + 1 snapshot

performance.reqs = 0 # no background read requests pending

performance.cursor_latency = 37845437 0 206111877 # avg close time is 37 seconds
```

**Comment by rtsisyk** [comment]
Discussed on-site. Topic starter wants an explicit method in Lua iterators to free memory, e.g. iterator:close() or iterator:recycle(). I agree that each Lua userdata/cdata should have such method to clean resources before waiting for unpredictable GC.

**Comment by rtsisyk** [comment]
This problem can be partially fixed by implicitly calling iterator_close after last tuple.

**Comment by rtsisyk** [comment]
OMG, iterator->close() was removed by b5e873073e4c6c42bad1dcbe815c502eddd80d31.

Could you please cleanup sophia resources just before returning NULL (=end of data) from iterator->next? 

```
struct tuple *
sophia_iterator_next(struct iterator *ptr)
{
	assert(ptr->next == sophia_iterator_next);
        ....
	obj = fiber_get_key(fiber(), FIBER_KEY_MSG);
	if (obj == NULL)
                /* No more data in iterator, cleanup Sophia resources here */
		return NULL; 
```

This workaround should help in the most common caseю

**Comment by kostja** [comment]
We can close the iterator automatically when the end is reached, this way we don't need an explicit method.

**Comment by pmwkaa** [comment]
The fix made as a part of #1318 ticket. Automatically close the cursor after last statement has been read.

**Comment by pmwkaa** [comment]
fixed in #1318
